### PR TITLE
feat: more adaptive content paths

### PR DESCRIPTION
**Pull request by ineshbose**
Original PR: #692

### CODE REVIEW

File: src/resolvers.ts [hunk 1]
```diff
@@ -22,17 +22,34 @@ export const resolveConfigPath = async (path: Arrayable<string>) => (
...
+      return []
+    })(),
+
+    r(`${nuxt.options.dir.layouts}/**/*${sfcExtensions}`),
+    ...(nuxt.options.pages ? [r(`${nuxt.options.dir.pages}/**/*${sfcExtensions}`)] : []),
+
+    r(`${nuxt.options.dir.plugins}/**/*${defaultExtensions}`),
+    r(`${nuxt.options.dir.modules}/**/*${defaultExtensions}`),
+    ...nuxt.options.modulesDir.map(m => r(`${m}/**/*${defaultExtensions}`)),
+    ...(nuxt.options.imports.dirs || []).map(d => r(`${d}/**/*${defaultExtensions}`)),
```
Comment (maddysrc) [inline src/resolvers.ts:46]: Suppose I'll move my comment to the new PR as well. `imports.dirs` doesn't end up including composables and utils as they're added by nuxt regardless of what the config value is.

**Review by maddysrc — CHANGES_REQUESTED** [review]

File: src/resolvers.ts (hunk 1 above)
Comment (ineshbose) [reply src/resolvers.ts:46]: How do you feel about this:

```ts

  const importDirs = [...(nuxt.options.imports.dirs || [])]
  const [composablesDir, utilsDir] = [resolve(srcDir, 'composables'), resolve(srcDir, 'utils')]

  if (!importDirs.includes(composablesDir)) importDirs.push(composablesDir)
  if (!importDirs.includes(utilsDir)) importDirs.push(utilsDir)

  // later..
 ...importDirs.map(d => r(`${d}/**/*${defaultExtensions}`)),
```

Comment (ineshbose) [reply src/resolvers.ts]: (committed)

File: src/resolvers.ts [hunk 2]
```diff
@@ -22,17 +22,40 @@ export const resolveConfigPath = async (path: Arrayable<string>) => (
...
+      }
+      return []
+    })(),
+
+    r(`${nuxt.options.dir.layouts}/**/*${sfcExtensions}`),
+    ...(nuxt.options.pages ? [r(`${nuxt.options.dir.pages}/**/*${sfcExtensions}`)] : []),
+
+    r(`${nuxt.options.dir.plugins}/**/*${defaultExtensions}`),
+    r(`${nuxt.options.dir.modules}/**/*${defaultExtensions}`),
+    ...nuxt.options.modulesDir.map(m => r(`${m}/**/*${defaultExtensions}`)),
```
Comment (maddysrc) [inline src/resolvers.ts:51]: I noticed that `modulesDir` actually refers to node_modules, probably good to drop that.

File: src/resolvers.ts (hunk 1 above)
Comment (maddysrc) [reply src/resolvers.ts]: `imports.dirs` should probably be mapped to an absolute path before attempting to deduplicate. Might also want to use a Set instead.

Something like this:
```js
const importDirs = new Set([...(nuxt.options.imports.dirs || [])].map(it => resolve(it)))
importDirs.add(resolve(srcDir, 'composables'))
importDirs.add(resolve(srcDir, 'utils'))
```

**Review by maddysrc — APPROVED** [review]

**Comment by ineshbose** [comment]
Thanks for reviewing @maddysrc 😄

**Comment by maddysrc** [comment]
When is this expected to be merged/released? Is anything blocking it?

**Comment by ineshbose** [comment]
When @Atinux reviews and approves (should be by Wednesday 01 Nov) 🙂 

Meanwhile if you can, would you also like to review #743 (planned to be included in same release 6.9.0)?

**Comment by Atinux** [comment]
Could you add a test case to see the generated paths please?

**Review by Atinux — APPROVED** [review]

**Pull request merged**